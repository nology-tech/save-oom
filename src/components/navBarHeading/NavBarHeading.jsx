import React from 'react';
import './NavBarHeading.scss';
import vector from '../../assets/Vectorvector.png';

const NavBarHeading = () => {
  return (
    <div className='nav-bar'>
      <h2 className='nav-bar__heading'>Phonics stage</h2>
      <img className='nav-bar__img' src={vector} alt='vector img' />
    </div>
  );
};

export default NavBarHeading;
