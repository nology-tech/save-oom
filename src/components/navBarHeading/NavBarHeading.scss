@use '../../styles/reset' as *;
.nav-bar {
  &__heading {
    position: absolute;
    width: 49px;
    height: 29px;
    left: 28px;
    top: 158px;

    font-family: 'Inter';
    font-style: normal;
    font-weight: 400;
    font-size: 16px;
    line-height: 180%;

    /* identical to box height, or 29px */
    display: flex;
    align-items: center;

    color: $color-accent;
  }
  &__img {
    position: absolute;
    left: 10%;
    right: 10%;
    top: 20%;
    bottom: 25%;
  }
}
