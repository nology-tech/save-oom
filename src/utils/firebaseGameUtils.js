import { db } from "../firebase";
import {
  addDoc,
  collection,
  getDocs,
  query,
  where,
  orderBy,
  Timestamp,
} from "firebase/firestore";
/** 
 * Get all the rounds of a specific game and level for a given user, ordered by most recent first 
 @param {*} userId
  @param {String} game
*@returns a querySnap (document snapshot), returns a Promise.
*/

export const getGameRoundsForUser = async (id, game) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("gameId", "==", game),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};

/** 
   * Get all the correct round of a specific game and level for a given user, ordered by most recent first 
  @param {String} userId
  @param {String} gameId
  @param {Number} level  
  *@returns a promise with the query snapshot.
  */
export const getCorrectGameRoundsForUser = async (id, game, level) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("gameId", "==", game),
    where("level", "==", level),
    where("correct", "==", true),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};

/** Get all the incorrect rounds for a given user, ordered by most recent first 
  @param {String} userId
  *@returns  a promise with the query snapshot.
  */
export const getIncorrectGameRoundsForUser = async (id) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("correct", "==", false),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};

/** Get all the incorrect round of a specific game for a given user, ordered by most recent first 
  @param {String} userId
  @param {String} gameId
  @param {Number} level  
  *@returns  a promise with the query snapshot.
  */

export const getAllIncorrectGameRoundsForUser = async (id, game) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("gameId", "==", game),
    where("correct", "==", true),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};



/** Function that returns an array of rounds 
  @param {String} userId
  @param {String} gameId
  @param {Function} getGameRounds
  *@returns returs a promise.
  */
export const getArrayOfRounds = async (id, game, level, getGameRounds) => {
  const querySnap = await getGameRounds(id, game, level);
  const phonicsArr = querySnap.docs.map((doc) => {
    return doc.data().phonic;
  });
  return phonicsArr;
};

/** Function that returns an array of rounds 
  @param {String} id
  @param {String} game
  @param {Object} roundInfo
  @param {String} answer
  *@returns returs a promise.
  */

export const saveUserRound = async (id, game, level, gameStats, answer) => {
  const { isCorrect, score } = gameStats;
  await addDoc(collection(db, "users", id, "rounds_played"), {
    correct: isCorrect,
    gameId: game,
    phonic: answer,
    level: level,
    roundTime: Timestamp.fromDate(new Date()).toDate(),
    score: score,
  });
};
