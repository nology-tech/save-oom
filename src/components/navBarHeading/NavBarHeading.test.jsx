import {render, screen} from "@testing-library/react"
import NavBarHeading from "./NavBarHeading";
import { customRender } from '../../utils/testUtils';


it('Should render the navbar header', () => {
    const { container } = customRender(<NavBarHeading />);
    expect(container).toMatchSnapshot();
});

it('Should contain the navbar heading', () => {
    render(<NavBarHeading />);
  
});