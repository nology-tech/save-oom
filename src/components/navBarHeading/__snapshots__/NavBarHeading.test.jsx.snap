// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the navbar header 1`] = `
<div>
  <div
    class="undefined-header"
  >
    <h2
      class="undefined-header__heading"
    />
  </div>
</div>
`;
